src/backend/distributed/planner/multi_logical_optimizer.c

```diff
@@ -4445,7 +4444,18 @@ FindReferencedTableColumn(Expr *columnExpression, List *parentQueryList, Query *
 	 */
 	if (candidateColumn->varlevelsup > 0)
 	{
-		return;
+		int parentQueryIndex = list_length(parentQueryList) -
```

Comment:
just to be on the safe side, we should check the length of `parentQueryList`. Users might call this function on an unexpected state (e.g., candidateColumn->varlevelsup > 0 AND parentQueryList = NIL)